python/ray/_private/utils.py

```diff
@@ -413,133 +333,23 @@ def set_omp_num_threads_if_unset() -> bool:
-    if last_set_tpu_chips == tpu_chips:
-        return  # optimization: already set
-    if len(tpu_chips) == ray_constants.RAY_TPU_NUM_CHIPS_PER_HOST:
-        # Let the ML framework use the defaults
-        return
-    _set_visible_ids(tpu_chips, ray_constants.TPU_VISIBLE_CHIPS_ENV_VAR)
-    num_chips = len(tpu_chips)
-    if num_chips == 1:
-        os.environ[
-            ray_constants.TPU_CHIPS_PER_HOST_BOUNDS_ENV_VAR
-        ] = ray_constants.TPU_CHIPS_PER_HOST_BOUNDS_1_CHIP_CONFIG
-        os.environ[
-            ray_constants.TPU_HOST_BOUNDS_ENV_VAR
-        ] = ray_constants.TPU_SINGLE_HOST_BOUNDS
-    elif num_chips == 2:
-        os.environ[
-            ray_constants.TPU_CHIPS_PER_HOST_BOUNDS_ENV_VAR
-        ] = ray_constants.TPU_CHIPS_PER_HOST_BOUNDS_2_CHIP_CONFIG
-        os.environ[
-            ray_constants.TPU_HOST_BOUNDS_ENV_VAR
-        ] = ray_constants.TPU_SINGLE_HOST_BOUNDS
-    last_set_tpu_chips = tpu_chips
-
-
-def _set_visible_ids(visible_ids: List[str], env_var: str):
-    """Set the environment variable (e.g., CUDA_VISIBLE_DEVICES, NEURON_RT_VISIBLE_CORES,
-     TPU_VISIBLE_CHIPS) passed based on accelerator runtime and will raise an error if
-     the function uses different environment variable.
+last_set_visible_accelerator_ids = {}
 
-    Args:
-        visible_ids (List[str]): List of strings representing GPU IDs or NeuronCore IDs.
-        env_var: Environment variable to set based on accelerator runtime.
 
+def set_visible_accelerator_ids() -> None:
+    """Set (CUDA_VISIBLE_DEVICES, NEURON_RT_VISIBLE_CORES, TPU_VISIBLE_CHIPS ,...)
+    environment variables based on the accelerator runtime.
     """
-    if env_var not in (
-        ray_constants.CUDA_VISIBLE_DEVICES_ENV_VAR,
-        ray_constants.NEURON_RT_VISIBLE_CORES_ENV_VAR,
-        ray_constants.TPU_VISIBLE_CHIPS_ENV_VAR,
+    global last_set_visible_accelerator_ids
+    for resource_name, accelerator_ids in (
+        ray.get_runtime_context().get_resource_ids().items()
     ):
-        raise ValueError(f"Invalid environment variable {env_var} to set visible IDs.")
-    os.environ[env_var] = ",".join([str(i) for i in visible_ids])
-
-
-def get_neuron_core_constraint_name():
-    """Get the name of the constraint that represents the AWS Neuron core accelerator.
-
-    Returns:
-        (str) The constraint name.
-    """
-    import ray.util.accelerators.accelerators as accelerators
-
-    return get_constraint_name(accelerators.AWS_NEURON_CORE)
-
-
-def get_constraint_name(pretty_name: str):
-    """Get the name of the constraint that represents the given resource.
-
-    Args:
-        pretty_name: The name of the resource.
-
-    Returns:
-        (str) The constraint name.
-    """
-    constraint_name = f"{ray_constants.RESOURCE_CONSTRAINT_PREFIX}" f"{pretty_name}"
-    return constraint_name
+        if last_set_visible_accelerator_ids.get(resource_name, None) == accelerator_ids:
+            continue  # optimization: already set
```

Comment:
Is it new or did we have the same optimization before?

Reply:
also is it really necessary? Aren't they just setting an env var? Maybe remove this for now?

Reply:
Same old code.

I actually tried to remove it since I also think it might be unnecessary but it uncovered a bug. I decided to fix the bug in a follow-up PR and then remove this optimization.